### ISSUE TITLE
tf.nn.max_pool3d crashes(floating point exception) when strides=0

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#42213]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.6.9
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**

`tf.nn.max_pool3d` crashes(floating point exception) when `input` is of type `np.float16` **AND** `strides=0` **AND** `padding='SAME'`

**Describe the expected behavior**
expect no crashes

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
~~~python
import tensorflow as tf
import numpy as np
input = tf.ones((1,1,1,1,1), dtype=np.float16)
tf.nn.max_pool3d(input=input, ksize=1, strides=0, padding='SAME')
~~~
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
~~~python
Floating point exception (core dumped)
~~~

**Comment by ravikyram** [comment]
@DNXie 

I tried in colab and i am not seeing any issue if we give give `stride >0` in code.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/a7183951523b0c440a894f53267c1e80/untitled237.ipynb).
but  `tf.nn.max_pool3d` crashes(floating point exception) when input is of type np.float16 AND strides=0 AND padding='SAME'.

Thanks!

**Comment by ravikyram** [comment]
I am able to reproduce the issue in colab with TF 2.3, nightly versions(`2.4.0-dev20200810`).Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/b1538e804cb9ede63700c745e895c179/untitled238.ipynb).Thanks!

**Comment by rmothukuru** [comment]
@DNXie,
It works fine if you use [Keras Mixed Precision Experimental Policy](https://www.tensorflow.org/api_docs/python/tf/keras/mixed_precision/experimental/Policy#how_to_use_mixed_precision_in_a_keras_model_2). Please find the [Gist with working code](https://colab.research.google.com/gist/rmothukuru/d4ceabd60b2b82930aa2aa5a6b914dc0/untitled238.ipynb). Thanks!

**Comment by rmothukuru** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by DNXie** [comment]
@rmothukuru Thanks for the clarification! Though an exception sounds better than crash (segfault) :)
